libs/ngxtension/click-outside/src/click-outside.ts

```diff
@@ -0,0 +1,62 @@
+import {
+	Directive,
+	ElementRef,
+	EventEmitter,
+	inject,
+	Injectable,
+	NgZone,
+	Output,
+} from '@angular/core';
+
+import type { OnInit } from '@angular/core';
+import { injectDestroy } from 'ngxtension/inject-destroy';
+import { fromEvent, Subject, takeUntil } from 'rxjs';
+
+/*
+ * This service is used to detect clicks in the document.
+ * It is used by the clickOutside directive.
+ */
+@Injectable({ providedIn: 'root' })
+export class DocumentClickService {
+	click$ = new Subject<Event>();
+
+	constructor(ngZone: NgZone) {
+		ngZone.runOutsideAngular(() => {
+			fromEvent(document, 'click').subscribe(this.click$);
+		});
+	}
+}
+
+/*
+ * This directive is used to detect clicks outside the element.
+ *
+ * Example:
+ * <div (clickOutside)="close()"></div>
+ *
+ */
+@Directive({ selector: '[clickOutside]', standalone: true })
+export class ClickOutsideDirective implements OnInit {
```

Comment:
nit: I would remove `Directive` from the name of the directive, just `ClickOutside` is enough